src/cli/index.js

```diff
@@ -20,347 +20,352 @@ const net = require('net');
-commander.option('--prefer-offline', 'use network only if dependencies are not available in local cache');
-commander.option('--strict-semver');
-commander.option('--json', '');
-commander.option('--ignore-scripts', "don't run lifecycle scripts");
-commander.option('--har', 'save HAR output of network traffic');
-commander.option('--ignore-platform', 'ignore platform checks');
-commander.option('--ignore-engines', 'ignore engines check');
-commander.option('--ignore-optional', 'ignore optional dependencies');
-commander.option('--force', 'install and build packages even if they were built before, overwrite lockfile');
-commander.option('--skip-integrity-check', 'run install without checking if node_modules is installed');
-commander.option('--check-files', 'install will verify file tree of packages for consistency');
-commander.option('--no-bin-links', "don't generate bin links when setting up packages");
-commander.option('--flat', 'only allow one version of a package');
-commander.option('--prod, --production [prod]', '');
-commander.option('--no-lockfile', "don't read or generate a lockfile");
-commander.option('--pure-lockfile', "don't generate a lockfile");
-commander.option('--frozen-lockfile', "don't generate a lockfile and fail if an update is needed");
-commander.option('--link-duplicates', 'create hardlinks to the repeated modules in node_modules');
-commander.option('--global-folder <path>', 'specify a custom folder to store global packages');
-commander.option(
-  '--modules-folder <path>',
-  'rather than installing modules into the node_modules folder relative to the cwd, output them here',
-);
-commander.option('--cache-folder <path>', 'specify a custom folder to store the yarn cache');
-commander.option('--mutex <type>[:specifier]', 'use a mutex to ensure only one yarn instance is executing');
-commander.option('--emoji [bool]', 'enable emoji in output', process.platform === 'darwin');
-commander.option('-s, --silent', 'skip Yarn console logs, other types of logs (script output) will be printed');
-commander.option('--proxy <host>', '');
-commander.option('--https-proxy <host>', '');
-commander.option('--no-progress', 'disable progress bar');
-commander.option('--network-concurrency <number>', 'maximum number of concurrent network requests', parseInt);
-commander.option('--network-timeout <milliseconds>', 'TCP timeout for network requests', parseInt);
-commander.option('--non-interactive', 'do not show interactive prompts');
-commander.option('--scripts-prepend-node-path [bool]', 'prepend the node executable dir to the PATH in scripts');
-
-// get command name
-let commandName: string = args.shift() || 'install';
-
-// if -v is the first command, then always exit after returning the version
-if (commandName === '-v') {
-  console.log(version.trim());
-  process.exitCode = 0;
-}
+export default function main() {
+  loudRejection();
+  handleSignals();
+
+  const startArgs = process.argv.slice(0, 2);
```

Comment:
What do you think about modularizing and putting this in the `main` parameters? I can see this function as an interesting entry point to run commands in our testcases - we currently need to instanciate each class and pass them the right properties, which might be tedious and error-prone, so it could be nice to just be able to use `run([ 'install', '--check-files' ])` instead and use the actual CLI pipeline.